File: 백준/Silver/1874. 스택 수열/스택 수열.js
const input = require('fs').readFileSync(0).toString().trim().split('\n').map(Number);

const n = input.shift();
const stack = [];
const answer = [];

let currentNumber = 1;
let sequence = true;

input.forEach((sequenceNumber) => {
  while (currentNumber <= sequenceNumber) {
    stack.push(currentNumber);
    answer.push('+');

    currentNumber += 1;
  }

  if (sequenceNumber === stack[stack.length - 1]) {
    stack.pop();
    answer.push('-');
  } else sequence = false;
});

if (sequence) console.log(answer.join('\n'));
else console.log('NO');

File: 백준/Silver/1874. 스택 수열/스택 수열.py
import sys
input = sys.stdin.readline

n = int(input())

stack = []
answer = []
sequence = True

currentNumber = 1
for _ in range(n):
  sequenceNumber = int(input())
  
  while currentNumber <= sequenceNumber:
    stack.append(currentNumber)
    answer.append('+')
    currentNumber += 1
  
  if stack[-1] == sequenceNumber:
    stack.pop()
    answer.append('-')

  else: 
    sequence = False

if sequence:
  for i in answer:
    print(i)
else:
  print('NO')